interpreter/src/vm/vm.c

```diff
@@ -204,7 +219,8 @@ bool ovm_engine_memory_ensure_capacity(ovm_engine_t *engine, i64 minimum_size) {
     if (engine->memory == NULL) {
         new_addr = mmap(NULL, minimum_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
     } else {
-        new_addr = mremap(engine->memory, engine->memory_size, minimum_size, MREMAP_MAYMOVE);
+        munmap(engine->memory, engine->memory_size);
+        new_addr = mmap(NULL, minimum_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
```

Comment:
Does this maintain the same behavior as `mremap`?

Reply:
I messed up and deleted the wrong thing... So I will retype my comment from memory.

I don't think that is the same thing. That will lose all current memory, and replace it with with a fresh set of zeroed-pages.
I think the closest solution would be:
```c
new_addr = mmap(engine->memory, minimum_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
if (new_addr == MAP_FAILED) {
    // Panic
    return false;
} else if (new_addr != engine->memory) {
    // Mapped region has moved, need to copy everything
    memcpy(new_addr, engine->memory, engine->memory_size);
    munmap(engine->memory);
} else {
    // Mapped more memory with the same start pointer, do nothing
}
```

Reply:
However, I think still using `mremap` on Linux would be better because it is guaranteed to work and be fast.

Reply:
🤦 can't believe I forgot the memcpy... Will update this

Reply:
Fixed and tested. Created  program that would segfault with the original changes, no longer does!

Reply:
Awesome!